packages/pools/src/concentrated/fetch-pool-amount-provider.ts

```diff
@@ -0,0 +1,71 @@
+import { Int } from "@keplr-wallet/unit";
+
+import { AmountsDataProvider, ConcentratedLiquidityPool } from "./pool";
+
+type BankAmountResponse = {
+  balances: {
+    amount: string;
+    denom: string;
+  }[];
+};
+
+/** Provider of bank balances from bank module by bech32Address. Expected by concentrated liquidity pool. */
+export class FetchPoolAmountProvider implements AmountsDataProvider {
+  /** Address => Response */
+  protected amountsCache = new Map<
+    string,
+    Awaited<ReturnType<AmountsDataProvider["getPoolAmounts"]>>
+  >();
+
+  constructor(
+    protected readonly baseNodeUrl: string,
+    protected readonly poolId: string,
+    protected readonly cacheDurationMs = 1000 * 30, // 30 seconds
```

Comment:
Our cache gets cleared only every 30 seconds. At scale, this could be an issue. I wonder if it would be worthwhile to implement something more sophisticated here such as LRU cache. https://www.npmjs.com/package/lru-cache

Reply:
We could look into that if it's an issue in the future, IMO. It would be a premature optimization and would increase the bundle size of the worker.

Reply:
I'll defer to you here since I don't really have a good sense of the use case here. I think the point about premature optimization is stronger than bundle size. If we are getting a lot of unique requests and our cache hit rate is low, than downloading the lru-cache package would be a clear solution.